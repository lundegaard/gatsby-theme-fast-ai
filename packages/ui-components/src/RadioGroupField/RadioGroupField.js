import React from 'react';
import PropTypes from 'prop-types';
import { isString } from 'ramda-extension';

import getEventHandlersFromProps from '../utils/getEventHandlersFromProps';
import RadioGroup from '../RadioGroup';
import FormGroup from '../FormGroup';

const RadioGroupField = ({
	children,
	onChange,
	value,
	disabled,
	readOnly,
	hasError,
	hint,
	legend,
	name,
	...rest
}) => (
	<FormGroup
		legend={legend}
		hint={hint}
		disabled={disabled}
		readOnly={readOnly}
		hasError={hasError}
		{...rest}
	>
		<RadioGroup
			aria-label={isString(legend) ? legend : null}
			onChange={onChange}
			value={value}
			disabled={disabled}
			readOnly={readOnly}
			hasError={hasError}
			name={name}
			{...getEventHandlersFromProps(rest)}
		>
			{children}
		</RadioGroup>
	</FormGroup>
);

RadioGroupField.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Disabled state */
	disabled: PropTypes.bool,
	/** Error state */
	hasError: PropTypes.bool,
	hint: PropTypes.node,
	legend: PropTypes.node,
	name: PropTypes.string,
	/** Called when `onChange` event is triggered on any of children which has `value` prop set. */
	onChange: PropTypes.func,
	/** Read-only state */
	readOnly: PropTypes.bool,
	/** Current value of the radio group. */
	value: PropTypes.any,
};

export default RadioGroupField;
