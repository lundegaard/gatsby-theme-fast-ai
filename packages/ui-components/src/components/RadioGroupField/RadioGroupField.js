import React from 'react';
import PropTypes from 'prop-types';
import { isString } from 'ramda-extension';

import Col from '../Col';
import Row from '../Row';
import Box from '../Box';
import Label from '../Label';
import RadioGroup from '../RadioGroup';
import SuperFieldHint from '../SuperFieldHint';

const getVariant = ({ hasError, readOnly, disabled }) => {
	if (hasError) {
		return 'error';
	}
	if (disabled || readOnly) {
		return 'disabled';
	}
};

const RadioGroupField = ({
	children,
	onChange,
	value,
	disabled,
	readOnly,
	hasError,
	hint,
	legend,
	// eslint-disable-next-line react/prop-types,no-unused-vars
	name,
	...rest
}) => (
	<Box
		as="fieldset"
		variant={getVariant({ hasError, readOnly, disabled })}
		sx={{ border: 'none', appearance: 'none', p: 0 }}
	>
		{/* https://stackoverflow.com/questions/28078681/why-cant-fieldset-be-flex-containers */}
		<Row flexWrap="wrap" {...rest}>
			<Col span={[12, 6, 5, 4]} pb={[2, 0]}>
				<Label as="legend" sx={{ width: '100%', p: 0 }}>
					{legend}
				</Label>
			</Col>
			<Col span={[12, 6, 7, 8]}>
				<RadioGroup
					aria-label={isString(legend) ? legend : null}
					onChange={onChange}
					value={value}
					disabled={disabled}
					readOnly={readOnly}
					hasError={hasError}
				>
					{children}
				</RadioGroup>
			</Col>
		</Row>
		<Row>
			<Col span={12}>{hint ? <SuperFieldHint>{hint}</SuperFieldHint> : null}</Col>
		</Row>
	</Box>
);

RadioGroupField.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Disabled state */
	disabled: PropTypes.bool,
	/** Error state */
	hasError: PropTypes.bool,
	hint: PropTypes.node,
	legend: PropTypes.node,
	/** Called when `onChange` event is triggered on any of children which has `value` prop set. */
	onChange: PropTypes.func,
	/** Read-only state */
	readOnly: PropTypes.bool,
	/** Current value of the radio group. */
	value: PropTypes.any,
};

export default RadioGroupField;
