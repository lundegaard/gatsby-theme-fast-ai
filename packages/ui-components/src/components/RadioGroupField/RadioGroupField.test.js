import React from 'react';
import { shallow } from 'enzyme';

import RadioGroupField from './RadioGroupField';

describe('<RadioGroupField />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<RadioGroupField {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
