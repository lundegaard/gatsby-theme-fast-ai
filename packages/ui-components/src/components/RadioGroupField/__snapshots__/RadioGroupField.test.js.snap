// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroupField /> matches snapshot when shallowly rendered 1`] = `
<Box
  as="fieldset"
  sx={
    Object {
      "appearance": "none",
      "border": "none",
      "p": 0,
    }
  }
>
  <Row
    flexWrap="wrap"
  >
    <Col
      pb={
        Array [
          2,
          0,
        ]
      }
      span={
        Array [
          12,
          6,
          5,
          4,
        ]
      }
    >
      <Label
        as="legend"
        sx={
          Object {
            "p": 0,
            "width": "100%",
          }
        }
      />
    </Col>
    <Col
      span={
        Array [
          12,
          6,
          7,
          8,
        ]
      }
    >
      <RadioGroup
        aria-label={null}
      />
    </Col>
  </Row>
  <Row>
    <Col
      span={12}
    />
  </Row>
</Box>
`;
